src/platform/datapath_winkernel.c

```diff
@@ -1552,6 +1600,47 @@ SocketCreateUdp(
         }
     }
 
+    if ((Datapath->Features & CXPLAT_DATAPATH_FEATURE_TYPE_OF_SERVICE) &&
+        Config->TypeOfService != 0) {
+        Option = Config->TypeOfService;
+        Status =
+            CxPlatDataPathSetControlSocket(
```

Comment:
We're going to have to use ancillary data instead, because of the server side. Server side shares the socket and some datagrams will be different traffic class.